pkg/odo/cli/component/common_push.go

```diff
@@ -113,7 +113,7 @@ func (cpo *CommonPushOptions) createCmpIfNotExistsAndApplyCmpConfig(stdout io.Wr
 				cmpName,
 				err,
 			)
-			os.Exit(1)
+			return err
```

Comment:
I think you should call `util.LogErrorAndExit(err, "")` here

Reply:
I am not sure why we want to use `LogErrorAndExit` here we are already doing logging in the above statement. And the error which is returned it just check if it's empty or not, if it's empty then it just does `return`.

Reply:
>I think you should call util.LogErrorAndExit(err, "") here

Why do you say that @feloy? I think the whole point of this PR is to avoid abrupt exit calls, and calling `LogErrorAndExit` defies that purpose in my opinion.

Reply:
@valaparthvi my bad, I misunderstood this function. I understood there had an existing function to be called that were exiting gracefully, I probably misunderstood